airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/poetry/pipeline.py

```diff
@@ -0,0 +1,209 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import configparser
+import io
+import os
+from textwrap import dedent
+from typing import Optional
+
+import tomli
+import tomli_w
+import yaml
+from pipelines.airbyte_ci.connectors.context import ConnectorContext, PipelineContext
+from pipelines.models.contexts.pipeline_context import PipelineContext
+from pipelines.models.steps import Step, StepResult
+
+
+class PyPIPublishContext(PipelineContext):
+    def __init__(
+        self,
+        pypi_token: str,
+        test_pypi: bool,
+        package_path: str,
+        build_docker_image: str,
+        ci_report_bucket: str,
+        report_output_prefix: str,
+        is_local: bool,
+        git_branch: bool,
+        git_revision: bool,
+        gha_workflow_run_url: Optional[str] = None,
+        dagger_logs_url: Optional[str] = None,
+        pipeline_start_timestamp: Optional[int] = None,
+        ci_context: Optional[str] = None,
+        ci_gcs_credentials: str = None,
+        package_name: Optional[str] = None,
+        version: Optional[str] = None,
+    ):
+        self.pypi_token = pypi_token
+        self.test_pypi = test_pypi
+        self.package_path = package_path
+        self.package_name = package_name
+        self.version = version
+        self.build_docker_image = build_docker_image
+
+        pipeline_name = f"Publish PyPI {package_path}"
+
+        super().__init__(
+            pipeline_name=pipeline_name,
+            report_output_prefix=report_output_prefix,
+            ci_report_bucket=ci_report_bucket,
+            is_local=is_local,
+            git_branch=git_branch,
+            git_revision=git_revision,
+            gha_workflow_run_url=gha_workflow_run_url,
+            dagger_logs_url=dagger_logs_url,
+            pipeline_start_timestamp=pipeline_start_timestamp,
+            ci_context=ci_context,
+            ci_gcs_credentials=ci_gcs_credentials,
+        )
+
+    @staticmethod
+    async def from_connector_context(connector_context: ConnectorContext) -> Optional["PyPIPublishContext"]:
+        """
+        Create a PyPIPublishContext from a ConnectorContext.
+
+        The metadata of the connector is read from the current workdir to capture changes that are not yet published.
+        If pypi is not enabled, this will return None.
+        """
+
+        current_metadata = yaml.safe_load(
```

Comment:
This doesn't work because `connector_context.connector.metadata` is what got published the last time, not the current state in the repo, so it wouldn't publish on the first merge that sets `remoteRegistries.pypi.enabled` to true.

Reply:
Good point, I could have sworn I ran into problems with this, that's why I changed it, but maybe I confused myself. I will look into this.